pkg/sql/distsql_plan_changefeed_test.go

```diff
@@ -687,3 +687,46 @@ func copyColumnAs(t *testing.T, table catalog.TableDescriptor, from, to tree.Nam
 	desc.SystemColumnKind = catpb.SystemColumnKind_NONE
 	return dst
 }
+
+// TestChangefeedPlanWithEmptyTable verifies that PlanCDCExpression can handle
+// tables with no columns without causing the "unable to determine result columns" error.
+// This test specifically targets a fix for the SQL error [XX000]:
+// "ERROR: internal error: unable to determine result columns" which occurs
+// when a CDC expression is planned against a table without columns.
+func TestChangefeedPlanWithEmptyTable(t *testing.T) {
```

Comment:
Could you also add a test that creates a changefeed on an empty table and validates getting rows? `TestChangefeedDiff` is an example of a such a test.

You could consider using something like this to get the row keys:
```
INSERT INTO empty DEFAULT VALUES RETURNING rowid;
```

Reply:
Test added